odoo/tests/common.py

```diff
@@ -835,25 +835,23 @@ def url_open(self, url, data=None, timeout=10):
         return self.opener.get(url, timeout=timeout)
 
     def _wait_remaining_requests(self):
-        t0 = int(time.time())
-        for thread in threading.enumerate():
-            if thread.name.startswith('odoo.service.http.request.'):
-                join_retry_count = 10
-                while thread.isAlive():
-                    # Need a busyloop here as thread.join() masks signals
-                    # and would prevent the forced shutdown.
-                    thread.join(0.05)
+        join_retry_count = 10
+        while join_retry_count:
+            threads = [t for t in threading.enumerate() if t.name.startswith('odoo.service.http.request')]
+            if not threads:
+                return
+            for t in threads:
+                t.join(0.05)
+                if t.isAlive():
                     join_retry_count -= 1
```

Comment:
This is going to decrement by 1 for every thread we've found alive, I'm guessing it's not going to go through 10 loops.

And since threading.enumerates() [only returns alive threads](https://docs.python.org/3/library/threading.html?highlight=threading#threading.enumerate), I don't know that we care about this: if we find living http threads, log that we're waiting on them, then decrement the counter and wait half a second.

Incidentally, since we're not using join_retry_count for anything except as a counter, we could just iterate on `range(10)` or `range(10, 0, -1)`, that way there's no need to (remember) to explicitly inc/dec the counter.

Reply:
The counter is decremented **only** if the thread is still alive after the join.
The idea is that if there is still a thread after 5sec, we have a problem hence the stack dump.